modules/gmake/gmake.lua

```diff
@@ -0,0 +1,338 @@
+
+	table.sort(wks.configs, function(cfg0, cfg1)
+		return (cfg0.gmake_identifier:lower() < cfg1.gmake_identifier:lower())
+	end)
+
+	return wks
+end
+
+
+---
+-- Fetch the settings for a specific project by name, adding values required by
+-- the Visual Studio exporter methods. Also fetches the configurations used by
+-- the project.
+--
+-- @param wks
+--	The `dom.Workspace` instance which contains the target project.
+-- @param name
+--	The name of the project to fetch.
+-- @returns
+--	A `dom.Project`, with additional Visual Studio specific values.
+---
+function gmake.fetchProject(wks, name)
+	local prj = dom.Project.new(wks
+		:select({ projects = name })
+		:fromScopes(wks.root)
+		:withInheritance()
+	)
+
+	prj.root = wks.root
+	prj.workspace = wks
+
+	local name = os.matchFiles(path.join(prj.location, 'Makefile')) and prj.name .. '.mak' or 'Makefile'
+	local prjPath = path.join(prj.location, name)
+
+	prj.exportPath = prjPath
+	prj.baseDirectory = path.getDirectory(prj.exportPath)
+	prj.uuid = prj.uuid or os.uuid(prj.name)
+
+	-- TODO: Compiler/Linker from DOM
+	prj.compiler = gmake.toolsets.compilers.gcc
+	prj.linker = gmake.toolsets.linkers.gcc
+
+	prj.configs = prj:fetchConfigs(gmake.fetchProjectConfig)
+
+	table.sort(prj.configs, function(cfg0, cfg1)
+		return (cfg0.gmake_identifier:lower() < cfg1.gmake_identifier:lower())
+	end)
+
+	return prj
+end
+
+
+---
+-- Fetch the settings for a specific workspace-level configuration.
+--
+-- @param wks
+--	The `dom.Workspace` instance which contains the target configuration.
+-- @param build
+--	The target build configuration name, eg. 'Debug'
+-- @param platform
+--	The target platform name.
+-- @returns
+--	A `dom.Config`, with additional Visual Studio specific values.
+---
+function gmake.fetchWorkspaceConfig(wks, build, platform)
+	local cfg = gmake.fetchConfig(wks
+		:selectAny({ configurations = build, platforms = platform })
+		:fromScopes(wks.root)
+		:withInheritance()
+	)
+
+	cfg.root = wks.root
+	cfg.workspace = wks
+
+	return cfg
+end
+
+
+---
+-- Fetch the settings for a specific project-level configuration.
+--
+-- @param wks
+--	The `dom.Project` instance which contains the target configuration.
+-- @param build
+--	The target build configuration name, eg. 'Debug'
+-- @param platform
+--	The target platform name.
+-- @returns
+--	A `dom.Config`, with additional Visual Studio specific values.
+---
+function gmake.fetchProjectConfig(prj, build, platform)
+	local cfg = gmake.fetchConfig(prj
+		:selectAny({ configurations = build, platforms = platform })
+		:fromScopes(prj.root, prj.workspace)
+	)
+
+	cfg.root = prj.root
+	cfg.workspace = prj.workspace
+	cfg.project = prj
+
+	-- TODO: Compiler/Linker from DOM
+	cfg.compiler = gmake.toolsets.compilers.gcc
+	cfg.linker = gmake.toolsets.linkers.gcc
+
+	-- Configurations inherit most values from project, but files should be kept separated
+	cfg.files = cfg:withoutInheritance().files
```

Comment:
Makefiles can actually support per-configuration file lists, unlike VS and Xcode. I'd like to see this exporter written that way, with common files listed at the project level, and only the files specific to each configuration listed in at the configuration level. Does that make sense? So you'd remove this and use the `cfg.files` that was already returned from the `fetchConfig()` call above.